alibuild_helpers/utilities.py

```diff
@@ -183,6 +183,17 @@ def readDefaults(configDir, defaults, error):
   if err:
     error(err)
     sys.exit(1)
+  archDefaults = "%s/defaults-%s.sh" % (configDir, architecture)
+  archMeta = {}
+  archBody = ""
+  if exists(archDefaults):
+    err, archMeta, archBody = parseRecipe(getRecipeReader(defaultsFilename))
+    if err:
+      error(err)
+      sys.exit(1)
+    for x in ["env", "disable", "overrides"]:
+      defaultsMeta.get(x, {}).update(archMeta.get(x, {}))
```

Comment:
This won't apply arch overrides from an `x` if `defaultsMeta` does not already have an `x` entry. Is that the desired behaviour? If not, the following might be better (works in Python 2 and 3):
```python
defaultsMeta.setdefault(x, {}).update(archMeta.get(x, {}))
```

Reply:
Nice catch.